app/main/views/suppliers.py

```diff
@@ -174,6 +174,30 @@ def approve_agreement_for_countersignature(agreement_id):
     ))
 
 
+@main.route('/suppliers/agreements/<agreement_id>/unapprove', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def unapprove_agreement_for_countersignature(agreement_id):
+    # not properly validating this - all we do is pass it through
+    next_status = request.args.get("next_status")
+
+    agreement = data_api_client.unapprove_agreement_for_countersignature(
+        agreement_id,
+        current_user.email_address,
+        current_user.id,
+    )["agreement"]
+    organisation = request.form['nameOfOrganisation']
+    flash(u'The agreement for {} had its approval cancelled. You can approve it again at any time.'
```

Comment:
fwiw I personally think that having flash messages in the templates is a bit of an anti-pattern; it begs the question "which template?" Because you want the flash messages to be able to appear everywhere, that basically means having some special "lookup table" template included from all other templates.